lib/rules/lint-require-button-type.js

```diff
@@ -0,0 +1,45 @@
+'use strict';
+
+const Rule = require('./base');
+
+const ERROR_MESSAGE = 'All `<button>` elements should have a valid `type` attribute';
+
+module.exports = class RequireButtonType extends Rule {
+  logNode({ node, message }) {
+    return this.log({
+      message,
+      line: node.loc && node.loc.start.line,
+      column: node.loc && node.loc.start.column,
+      source: this.sourceForNode(node),
+    });
+  }
+
+  visitor() {
+    return {
+      ElementNode(node) {
+        let { tag, attributes } = node;
+
+        if (tag !== 'button') {
+          return;
+        }
+
+        let typeAttribute = attributes.find(it => it.name === 'type');
+        if (!typeAttribute) {
+          return this.logNode({ node, message: ERROR_MESSAGE });
+        }
+
+        let { value } = typeAttribute;
+        if (value.type !== 'TextNode') {
+          return;
+        }
+
+        let { chars } = value;
+        if (!['button', 'submit', 'reset'].includes(chars)) {
+          return this.logNode({ node, message: ERROR_MESSAGE });
+        }
+      },
+    };
+  }
+};
+
+module.exports.ERROR_MESSAGE = ERROR_MESSAGE;
```

Comment:
What do you think of using a static class field to export ERROR_MESSAGE?

Reply:
I think that's essentially what the code is currently doing. I don't think static class fields are supported by all of our currently supported Node versions though.